openapi_tester/schema_tester.py

```diff
@@ -79,79 +67,40 @@ def test_dict(self, schema: dict, data: dict, reference: str) -> None:
         :param reference: string reference pointing to function caller
         :raises: openapi_tester.exceptions.DocumentationError
         """
-        if not isinstance(data, dict):
-            response_hint, request_hint = '', ''
-            if isinstance(data, list):
-                response_hint = 'The expected item should be a dict, or your schema should be a list.'
-                request_hint = 'You passed a list where the expected item should be a dict.'
-            elif data is None:
-                if 'x-nullable' in schema and schema['x-nullable']:
-                    # NoneTypes are OK if the schema says the field is nullable
-                    return
-                response_hint = (
-                    'If you wish to allow null values for this schema item, your schema needs to '
-                    'set `x-nullable: True`.\nFor drf-yasg implementations, set `x_nullable=True` '
-                    'in your Schema definition.'
-                )
-                request_hint = 'You passed a None value where we expected a dict.'
-            raise DocumentationError(
-                message=f"Mismatched types. Expected item to be <class 'dict'> but found {type(data)}.",
-                response=data,
-                schema=schema,
-                reference=reference,
-                response_hint=response_hint or '',
-                request_hint=request_hint or '',
-            )
```

Comment:
What will happen now if a response is documented as a dictionary, but contains some other type/is misdocumented?

Looks like it will fail a few lines down on line 113 `response_keys = data.keys()` instead.

Reply:
nope, we are now using prance to validate the schema in its entirety. If the schema is invalid it will fail completely upon load. Otherwise we are raising an "expected" vs. "received" error, which is what the user needs.